thunder/core/prims.py

```diff
@@ -2536,6 +2540,103 @@ def _uniform_meta(
 )
 
 
+def _set_rng_state_meta(new_state: TensorProxy, device: devices.Device | None = None) -> TensorProxy:
+    utils.check_type(new_state, TensorProxy)
+    utils.check(
+        isinstance(new_state.dtype, dtypes.unsignedinteger),
+        lambda: f"new_state dtype={new_state.dtype} was not a uint8 dtype",
+    )
+    utils.check(new_state.device is devices.cpu, lambda: f"new_state should be CPU tensor")
+    if device is not None:
+        utils.check(
+            device.devicetype is devices.DeviceType.CUDA,
+            lambda: f"set_rng_state is supported for CUDA only",
+            exception_type=NotImplementedError,
+        )
+    return TensorProxy(like=new_state)
+
+
+set_rng_state = make_prim(
+    PrimIDs.SET_RNG_STATE,
+    "set_rng_state",
+    meta=_set_rng_state_meta,
+    tags=(OpTags.RANDOM_OP, OpTags.DONT_DCE),
+)
+
+
+# NOTE: the input state is used to force the dependency when using multiple get/set_rng_state calls,
```

Comment:
I think we're making sure that the relative order of these calls is preserved under sorting through the dependency on the outputs of previous calls.
```suggestion
# NOTE: the input state is used to force relative ordering when using multiple get/set_rng_state calls,
```